### PR TITLE
Update toolchain to 1.79.0

### PR DESCRIPTION
**Pull request by roypat**
## Changes

Updates the rust toolchain to 1.79.0. Along the way, fixes various issues that come with dependency updates that we picked up when rebuilding the docker container (because we don't properly pin most of the things in there), such as 
- kani
- cargo-deny
- python's requests

Also make our "devctr sanity build" step trigger on poetry.lock changes, so that we notice ahead of the time when dependabot breaks stuff for us again (again, python's requests)

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license. For more information on following Developer
Certificate of Origin and signing off your commits, please check
[`CONTRIBUTING.md`][3].

## PR Checklist

- [ ] If a specific issue led to this PR, this PR closes the issue.
- [ ] The description of changes is clear and encompassing.
- [ ] Any required documentation changes (code and docs) are included in this
  PR.
- [ ] API changes follow the [Runbook for Firecracker API changes][2].
- [ ] User-facing changes are mentioned in `CHANGELOG.md`.
- [ ] All added/changed functionality is tested.
- [ ] New `TODO`s link to an issue.
- [ ] Commits meet
  [contribution quality standards](https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md#contribution-quality-standards).

______________________________________________________________________

- [ ] This functionality cannot be added in [`rust-vmm`][1].

[1]: https://github.com/rust-vmm
[2]: https://github.com/firecracker-microvm/firecracker/blob/main/docs/api-change-runbook.md
[3]: https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md

### CODE REVIEW

**Review by ShadowCurse — DISMISSED** [review]

**Review by pb8o — DISMISSED** [review]

File: tools/devctr/poetry.lock [hunk 1]
```diff
@@ -2,87 +2,87 @@
 
 [[package]]
 name = "aiohttp"
-version = "3.9.4"
+version = "3.9.5"
```
Comment (pb8o) [inline tools/devctr/poetry.lock:5]: It seems something is off with the update? The devctr sanity check seems to fail

Comment (roypat) [reply tools/devctr/poetry.lock:5]: Yeah, I kinda don't understand how this lockfile works tbh... if external factors can break it, then what is the point of having a lockfile? lol

Comment (roypat) [reply tools/devctr/poetry.lock:5]: fixed it by regenerating the lockfile

**Review by pb8o — APPROVED** [review]

**Review by zulinx86 — APPROVED** [review]
LGTM! Thank you for solving a bunch of related things (like offset_of)!!

**Pull request merged**